ambari-server/src/main/java/org/apache/ambari/server/controller/internal/RequestResourceProvider.java

```diff
@@ -324,6 +329,23 @@ public RequestStatusResponse invoke() throws AmbariException, AuthorizationExcep
     return resources;
   }
 
+  private void authorizeGetResources(String clusterName) throws NoSuchParentResourceException, AuthorizationException {
+    final boolean ambariLevelRequest = StringUtils.isBlank(clusterName);
+    final ResourceType resourceType = ambariLevelRequest ? ResourceType.AMBARI : ResourceType.CLUSTER;
+    Long resourceId;
+    try {
+      resourceId = ambariLevelRequest ? null : getClusterResourceId(clusterName);
+    } catch (AmbariException e) {
+      throw new NoSuchParentResourceException("Error while fetching cluster resource ID", e);
+    }
+    final Set<RoleAuthorization> requiredAuthorizations = ambariLevelRequest ? Sets.newHashSet(AMBARI_VIEW_STATUS_INFO)
```

Comment:
`EnumSet.of` is probably cheaper than `Sets.newHashSet`, but no matter.  All is good.